### ISSUE TITLE
Comparing function name

### ISSUE THREAD

**Issue by csw244** [Samsung/jalangi2#59]
Hello, I am testing for the code to check for the params that are passed in post functions. 
To do that, I wrote an analysis to check if the function call is post, but it is returning 'ReferenceError: post is not defined'. Below is the snippet of analysis that I wrote and the error is caused by 'var POST_FUNCTION= post'. Referring to the function name like this was found in Jalangi online demo. I understand that it is not used supported anymore, however is there any other way to check function's name in either invokeFunPre or invokeFun by comparing parameter f? If not, using any functions provided by jalangi?

ANALYSIS:
```
 var POST_FUNCTION = post;

this.invokeFunPre = function (iid, f, base, args, isConstructor) {
      console.log('function call intercepted before invoking');    
      if (f === POST_FUNCTION && args) {
      console.log('function is POST'); 
       // pass in config, always thrid parameter in http.post
        checkParams(args[2]);
      }
    };
```

I am also including the code that I am testing below.
CODE:
```
function post() {
} 

var sampleRecords = function(params) {
  var config = {
    params: params,
    httpErrorHandlers: {
      '4xx': function(error) {
        deferred.reject(error);
      }
    }
  }
      
post('/machinelearning/service/datasource/sample-records', null, config);
}
    
var params = {
  dbPassword : "password_test",
  dbUsername : "username_test"
};

sampleRecords(request);
```

**Comment by ksen007** [comment]
Declaration of the variable post happens after the analysis code is loaded.  Jalangi loads all runtime files and analysis files before loading any file under test.

**Comment by csw244** [comment]
In that case, what is the step to check the function name? Is it possible in Jalangi?

**Comment by ksen007** [comment]
In your analysis file, you can have some code that is executed once only after you have started executing the file under test.

var isFirst = true;

this.literal = function (iid, val, hasGetterSetter) {
   if (isFirst) {
        POST_FUNCTION = post;
        isFirst = false;
    }
        };
